src/Npgsql/NpgsqlTypes/NpgsqlRange.cs

```diff
@@ -131,49 +199,215 @@ internal NpgsqlRange([CanBeNull] T lowerBound, [CanBeNull] T upperBound, RangeFl
                x.LowerBoundInfinite == y.LowerBoundInfinite &&
                x.UpperBoundInfinite == y.UpperBoundInfinite;
 
+        /// <summary>
+        /// Indicates whether the <see cref="NpgsqlRange{T}"/> on the left is not equal to the <see cref="NpgsqlRange{T}"/> on the right.
+        /// </summary>
+        /// <param name="x">The <see cref="NpgsqlRange{T}"/> on the left.</param>
+        /// <param name="y">The <see cref="NpgsqlRange{T}"/> on the right.</param>
+        /// <returns>
+        /// True if the <see cref="NpgsqlRange{T}"/> on the left is not equal to the <see cref="NpgsqlRange{T}"/> on the right; otherwise, false.
+        /// </returns>
         public static bool operator !=(NpgsqlRange<T> x, NpgsqlRange<T> y) => !(x == y);
 
-        public override bool Equals([CanBeNull] object o) => o is NpgsqlRange<T> && this == (NpgsqlRange<T>)o;
+        /// <inheritdoc />
+        public override bool Equals(object o) => o is NpgsqlRange<T> range && this == range;
 
+        /// <inheritdoc />
         public bool Equals(NpgsqlRange<T> other) => this == other;
 
+        /// <inheritdoc />
         public override int GetHashCode()
-            => IsEmpty ? 0 :
-                (LowerBoundInfinite ? 0 : LowerBound.GetHashCode()) +
-                (UpperBoundInfinite ? 0 : UpperBound.GetHashCode());
+            => IsEmpty
+                ? 0
+                : (LowerBoundInfinite ? 0 : LowerBound.GetHashCode()) +
+                  (UpperBoundInfinite ? 0 : UpperBound.GetHashCode());
 
+        /// <inheritdoc />
```

Comment:
Unrelated, but I just realized I replicated this logic in the EF Core provider's [`NpgsqlRangeTypeMapping.GenerateNonNullSqlLiteral()`](https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/blob/14d4579c59f42c3319cfa39fc0bcd0adca6a3c8a/src/EFCore.PG/Storage/Internal/Mapping/NpgsqlRangeTypeMapping.cs#L61). Do you want to compare the two and submit a PR which makes EF Core simply call `ToString()`?

Reply:
Sure, I'll take care of that soon.

Reply:
See: https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/pull/420